src/components/timestamp-diff-util.jsx

```diff
@@ -0,0 +1,202 @@
+            //If their contents are identical
+            if (_.isEqual(del[k].innerHTML, ins[j].innerHTML)) {
+              try {
+                let dirtyDelXpath = xpath.fromNode(del[k], domDel);
+                let dirtyInsXpath = xpath.fromNode(ins[j], domIns);
+                let delxpath = removeDiffXPATH(dirtyDelXpath, 'del');
+                let insxpath = removeDiffXPATH(dirtyInsXpath, 'ins');
+                if (_.isEqual(delxpath, insxpath)) {
+                  deleteNodes(ins[j], del[k]);
+                  break;
+                } else {
+                  j--;
+                }
+              } catch (e) {
+                console.warn('Element might already be removed. Skipping..');
+                break;
+              }
+            }
+          }
+          j--;
+        }
+      }
+      k--;
+      j = ins.length - 1;
+    }
+  }
+  return {insertions: domIns.outerHTML, deletions: domDel.outerHTML};
+}
+
+function getLinkFromElement (hasLink) {
+  if (!_.isNil(hasLink.src))
+    return hasLink.src;
+  if (!_.isNil(hasLink.href))
+    return hasLink.href;
+  return hasLink.action;
+}
+
+function isNotAResource (element) {
+  return (!element.src && !element.href && !element.action);
+}
+
+function checkTimestampInLink (element) {
+  let link = getLinkFromElement(element);
+  if (_.isNil(link)) {
+    link = getLinkFromElement(element.parentNode.parentNode);
+  }
+  if (!_.isNil(link)) {
+    if (link.match(absoluteUrlRegex)) {
+      return element;
+    }
+    if (link.match(relativeUrlRegex)) {
+      return element;
+    }
+
+    element.childNodes.forEach(function (e) {
+      checkTimestampInLink(e);
+    });
+  }
+}
+
+function deleteNodes (nodeIns, nodeDel) {
+  removeMarkup(nodeIns);
+  removeMarkup(nodeDel);
+}
+
+function removeMarkup (node) {
+  if (node.className === 'wm-diff') {
+    node.outerHTML = node.innerHTML;
+  } else {
+    let parentNode = node.parentNode;
+    removeMarkup(parentNode);
+  }
+}
+
+function addNotNill (array, element) {
+  if (!_.isNil(element)) {
+    let url = getWBMCleanURL(element);
+    if (_.isNil(url)) {
+      url = getWBMCleanURL(element.parentNode.parentNode);
+    }
+    if (!_.isNil(url)) {
+      array.push([element, normalizeURL(url)]);
+    }
+  }
+}
+
+function getWBMCleanURL (element) {
+  if (!_.isNil(element.src))
+    return removeWBM(element.src);
+  if (!_.isNil(element.href))
+    return removeWBM(element.href);
+  if (!_.isNil(element.action))
+    return removeWBM(element.action);
+}
+
+function removeWBM (url){
+  let urlArray = url.split('/');
+  urlArray = urlArray.slice(6);
+  return urlArray.join('/');
+}
+function removeDiffXPATH (xpath, mode){
+  let xpathArray = xpath.split('/');
+  for (let i = 0; i < xpathArray.length; i++) {
+    if (xpathArray[i].includes(mode+'[')){
+      xpathArray.splice(i);
+      return xpathArray.join('/');
+    }
+  }
+}
+
+function normalizeURL (url) {
+  let lowercaseString = url.toLowerCase();
```

Comment:
Why do we need to lowercase?
There will be a problem, case matters in URL params. e.g. the following YT link breaks when we lowercase
https://www.youtube.com/watch?v=oku4Rk5IGIw